pages/login.page.js

```diff
@@ -0,0 +1,35 @@
+const PAGE_TIMEOUT = 10000;
+
+class LoginPage {
+    
+    get usernameInput () { return $('#username'); }
+    get passwordInput () { return $('#password'); }
+    get logInButton () { return $('#submit'); }
+    get rememberMe () { return $('#remember-me'); }
+    get failedLoginMessage () {return $('[class= "badge badge-pill badge-danger"]'); }
+
+
+    async completeFields(element, text) {
+        await element.waitForClickable({ timeout: PAGE_TIMEOUT });
+        await element.setValue(text);
+    }
+
+    async open(link) {
```

Comment:
There are methods that you are re-implementing on every page object (thus duplicating them) that are common to any page you are modeling: This is an example (`open()`), `completeFields()` and `clickElement()` are other examples.

Consider implementing behaviour that is common to every class in a super class (In page object lingo, that'd be something like a **BasePage**), and make every other page **inherit** from it, that way you only need to define those methods there, and pages than **inherits / extends** from it automatically get those methods by default, removing that duplication.